signals.go

```diff
@@ -17,6 +18,10 @@ const signalBufferSize = 2048
 // newSignalHandler returns a signal handler for processing SIGCHLD and SIGWINCH signals
 // while still forwarding all other signals to the process.
 func newSignalHandler(tty *tty) *signalHandler {
+	// set us as the subpreaper before registering the signal handler for the container
+	if err := system.SetSubreaper(1); err != nil {
```

Comment:
Yeah, should be safe, but I think it will be better to add a flag if someone wants to use their own higher level process as the reaper.